analyzers/src/SonarAnalyzer.CSharp/Rules/MultipleVariableDeclarationCodeFix.cs

```diff
@@ -32,54 +27,38 @@ public class MultipleVariableDeclarationCodeFix : MultipleVariableDeclarationCod
     {
         protected override SyntaxNode CalculateNewRoot(SyntaxNode root, SyntaxNode node)
         {
-            if (!(node is VariableDeclaratorSyntax declarator))
+            if (node is not VariableDeclaratorSyntax { Parent: VariableDeclarationSyntax declaration })
             {
                 return root;
             }
 
-            if (!(declarator.Parent is VariableDeclarationSyntax declaration))
-            {
-                return root;
-            }
-
-            var newDeclarations = declaration.Variables.Select(variable =>
-                SyntaxFactory.VariableDeclaration(
-                    declaration.Type.WithoutTrailingTrivia(),
-                    SyntaxFactory.SeparatedList(new[] { variable.WithLeadingTrivia(GetLeadingTriviaFor(variable)) })));
+            var newDeclarations = declaration.Variables.Select(variable => CreateNewDeclaration(variable, declaration));
```

Comment:
The `newNodes` looks too lonely with the empty lines around. At least the preceding could be removed